<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PCBS">
<title>Manipulating Percent Methylation • PCBS</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Manipulating Percent Methylation">
<meta property="og:description" content="PCBS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PCBS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Differential_Methylation.html">Differential Methylation with PCBS</a>
    <a class="dropdown-item" href="../articles/Manipulating_Percent_Methylation.html">Manipulating Percent Methylation</a>
    <a class="dropdown-item" href="../articles/Optimizing_DMR_Calling.html">Optimizing DMR Calling</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/katlande/PCBS/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Manipulating Percent Methylation</h1>
                        <h4 data-toc-skip class="author">Kathryn
Lande</h4>
            
            <h4 data-toc-skip class="date">2024-08-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/katlande/PCBS/blob/HEAD/vignettes/Manipulating_Percent_Methylation.Rmd" class="external-link"><code>vignettes/Manipulating_Percent_Methylation.Rmd</code></a></small>
      <div class="d-none name"><code>Manipulating_Percent_Methylation.Rmd</code></div>
    </div>

    
    
<p>While PCBS functions on eigenvector scores internally, users may wish
to generate plots using less esoteric values, such as the mean percent
methylation difference between conditions. Here we provide a brief
tutorial for this type of analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/katlande/PCBS" class="external-link">PCBS</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="generating-a-chromdict">Generating a chromDict<a class="anchor" aria-label="anchor" href="#generating-a-chromdict"></a>
</h2>
<p>chromDicts are lists of keyed data.tables that enable very fast
computing times. chromDicts of eigenvector score are used internally by
many PCBS functions. They can also be explicitly generated by users who
wish to speed up iterative function calls. In order to easily manipulate
our percent methylation values genome-wide, we can make a percent
methylation chromDict object that holds the average percent methylation
difference between two treatments for all filtered loci.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDictMethylDiff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/chromDictMeth.html" class="external-link">chromDictMeth</a></span><span class="op">(</span><span class="va">eigen</span>, IDs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"trt"</span>, <span class="st">"ctl"</span><span class="op">)</span>, filter_thresh <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="mean-methylation-over-regions">Mean methylation over regions<a class="anchor" aria-label="anchor" href="#mean-methylation-over-regions"></a>
</h2>
<p>Once the chromDictMeth object is generated, methylation at a region
or over a set of regions can be computed very quickly. Get the
methylation difference across one known DMR:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://katlande.github.io/PCBS/reference/MethylDiff.html" class="external-link">MethylDiff</a></span><span class="op">(</span><span class="va">chromDictMethylDiff</span>, <span class="st">"chr3"</span>, <span class="st">"4920450"</span>, <span class="st">"4923267"</span><span class="op">)</span></span>
<span><span class="co">## [1] -0.5493768</span></span></code></pre></div>
<p>Or get the methylation difference across a set of regions:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we need a data.frame of regions to check:</span></span>
<span><span class="co"># any dataframe where the first three column are chrom, start, end, respectively.</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr3"</span>, <span class="st">"chr1"</span><span class="op">)</span>,</span>
<span>                      s<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4920450</span>, <span class="fl">3961576</span>, <span class="fl">301000</span><span class="op">)</span>,</span>
<span>                      e<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4923267</span>, <span class="fl">3963805</span>, <span class="fl">303900</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/MethyDiff_Set.html" class="external-link">MethyDiff_Set</a></span><span class="op">(</span><span class="va">chromDictMethylDiff</span>, <span class="va">regions</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span> <span class="co"># returns the input file with a "mean_methylDiff" column</span></span>
<span><span class="co">##    chr       s       e mean_methylDiff</span></span>
<span><span class="co">## 1 chr3 4920450 4923267     -0.54937676</span></span>
<span><span class="co">## 2 chr3 3961576 3963805      0.21200740</span></span>
<span><span class="co">## 3 chr1  301000  303900     -0.02724904</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="metagenes---methylation">Metagenes - ∆ Methylation<a class="anchor" aria-label="anchor" href="#metagenes---methylation"></a>
</h2>
<p>Like the score_metagene() function, PCBS can also generate metagenes
on percent methylation using a chromDictMeth object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">chromDictMethylDiff</span>, <span class="va">regions</span><span class="op">)</span></span></code></pre></div>
<p><img src="Manipulating_Percent_Methylation_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Just like score-based metagenes, you can save the data from a
methylDiff_metagene() function and overlay multiple metagenes on a
single plot.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Some hypomethylated sites:</span></span>
<span><span class="va">hyper</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>c<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>                    s<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">8650728</span>, <span class="fl">3960576</span><span class="op">)</span>,</span>
<span>                    e<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">8652017</span>, <span class="fl">3962805</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Some hypermethylated sites:</span></span>
<span><span class="va">hypo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>c<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr3"</span>, <span class="st">"chr3"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>                   s<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4920450</span>, <span class="fl">4140014</span>, <span class="fl">1340221</span><span class="op">)</span>,</span>
<span>                   e<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4923267</span>, <span class="fl">4142951</span>, <span class="fl">1342791</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metagene_hyper</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">chromDictMethylDiff</span>, <span class="va">hyper</span>, return.data <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">metagene_hypo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">chromDictMethylDiff</span>, <span class="va">hypo</span>, return.data <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/multiple_metagenes.html">multiple_metagenes</a></span><span class="op">(</span>data_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">metagene_hyper</span>, <span class="va">metagene_hypo</span><span class="op">)</span>, <span class="co"># list of raw data</span></span>
<span>                   set_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hyper DMRs"</span>, <span class="st">"Hypo DMRs"</span><span class="op">)</span>, <span class="co"># names for elements of the data_list list</span></span>
<span>                   title<span class="op">=</span><span class="st">"Metagenes of DMR Regions"</span>, legend.title <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                   yaxis <span class="op">=</span> <span class="st">"Mean Methylation Difference"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Manipulating_Percent_Methylation_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="metagenes---other-values">Metagenes - Other Values<a class="anchor" aria-label="anchor" href="#metagenes---other-values"></a>
</h2>
<p>More broadly, we can make a chromDict of essentially any value, and
by extension, a metagene of essentially any value. The chromDict()
function only needs a single column dataframe, where genomic loci are
row.names. For example:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we need a data.frame of regions to check:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/getPCRanks.html">getPCRanks</a></span><span class="op">(</span><span class="va">eigen</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"trt"</span>, <span class="st">"ctl"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">##                   PC_Score</span></span>
<span><span class="co">## chr2:2505718  3.781088e-05</span></span>
<span><span class="co">## chr2:1807326  4.573342e-04</span></span>
<span><span class="co">## chr2:2490208  3.156102e-03</span></span>
<span><span class="co">## chr2:5512828 -5.695585e-03</span></span>
<span><span class="co">## chr3:1296629  2.822282e-03</span></span>
<span><span class="co">## chr3:3212276 -6.044066e-03</span></span></code></pre></div>
<p>Normally, this function takes the PC score, but we can also feed it
another value, such as total coverage:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we need a data.frame of regions to check:</span></span>
<span><span class="va">eigen</span><span class="op">$</span><span class="va">meanCounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"nCpG"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="co"># the column in our dict will be called "totalCounts"</span></span>
<span><span class="va">eigen</span> <span class="op">&lt;-</span> <span class="va">eigen</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"cpgID"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">13</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">##              meanCounts</span></span>
<span><span class="co">## chr2:2505718   56.50000</span></span>
<span><span class="co">## chr2:1807326   59.83333</span></span>
<span><span class="co">## chr2:2490208   57.33333</span></span>
<span><span class="co">## chr2:5512828   46.16667</span></span>
<span><span class="co">## chr3:1296629   55.83333</span></span>
<span><span class="co">## chr3:3212276   53.50000</span></span></code></pre></div>
<p>Then we can make a metagene of coverage:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we need a data.frame of regions to check:</span></span>
<span><span class="va">coverageDict</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chromDict.html">chromDict</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">13</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># metagenes on alternative values are made the same way, but the added "value" parameter must be specified. </span></span>
<span><span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">coverageDict</span>, <span class="va">hyper</span>, </span>
<span>                    yaxis <span class="op">=</span> <span class="st">"Total Sequencing Depth"</span>, <span class="co"># y-axis name is "Methylation Difference" automatically</span></span>
<span>                    value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">coverageDict</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="co"># column name for the plotted value inside your dict</span></span></code></pre></div>
<p><img src="Manipulating_Percent_Methylation_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>We can also make separate chromDicts for the treatment and control
samples. This way, we can plot the mean methylation in young and old
separately:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># matrix of percent methylation only:</span></span>
<span><span class="va">eigen</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"PercMeth"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">-&gt;</span> <span class="va">eigen_meth</span></span>
<span></span>
<span><span class="co"># take the mean of "trt" and "ctl" columns separately:</span></span>
<span><span class="va">eigen</span><span class="op">$</span><span class="va">trt_mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">eigen_meth</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"trt"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eigen_meth</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">eigen</span><span class="op">$</span><span class="va">ctl_mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">eigen_meth</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"ctl"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eigen_meth</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># make dicts for trt and ctl:</span></span>
<span><span class="va">trt_Dict</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chromDict.html">chromDict</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">[</span><span class="st">"trt_mean"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">ctl_Dict</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chromDict.html">chromDict</a></span><span class="op">(</span><span class="va">eigen</span><span class="op">[</span><span class="st">"ctl_mean"</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># make metagenes for each dict separately:</span></span>
<span><span class="va">metagene_trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">trt_Dict</span>, <span class="va">hyper</span>, return.data <span class="op">=</span> <span class="cn">T</span>, value <span class="op">=</span> <span class="st">"trt_mean"</span><span class="op">)</span></span>
<span><span class="va">metagene_ctl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://katlande.github.io/PCBS/reference/methylDiff_metagene.html" class="external-link">methylDiff_metagene</a></span><span class="op">(</span><span class="va">ctl_Dict</span>, <span class="va">hyper</span>, return.data <span class="op">=</span> <span class="cn">T</span>, value <span class="op">=</span> <span class="st">"ctl_mean"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot metagenes of % methylation at hypermeth sites in each condition separately:</span></span>
<span><span class="fu"><a href="../reference/multiple_metagenes.html">multiple_metagenes</a></span><span class="op">(</span>data_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">metagene_trt</span>, <span class="va">metagene_ctl</span><span class="op">)</span>, <span class="co"># list of raw data</span></span>
<span>                   set_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"trt"</span>, <span class="st">"ctl"</span><span class="op">)</span>, <span class="co"># names for elements of the data_list list</span></span>
<span>                   title<span class="op">=</span><span class="st">"% Methylation at Hyper DMRs"</span>, legend.title <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                   yaxis <span class="op">=</span> <span class="st">"Mean % Methylation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Manipulating_Percent_Methylation_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Kathryn Lande.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
